modules/indexer/code/bleve.go

```diff
@@ -333,13 +378,35 @@ func (b *BleveIndexer) Search(repoIDs []int64, keyword string, page, pageSize in
 				endIndex = locationEnd
 			}
 		}
+		language := hit.Fields["Language"].(string)
+		var updatedUnix timeutil.TimeStamp
+		if t, err := time.Parse(time.RFC3339, hit.Fields["UpdatedAt"].(string)); err == nil {
+			updatedUnix = timeutil.TimeStamp(t.Unix())
+		}
 		searchResults[i] = &SearchResult{
-			RepoID:     int64(hit.Fields["RepoID"].(float64)),
-			StartIndex: startIndex,
-			EndIndex:   endIndex,
-			Filename:   filenameOfIndexerID(hit.ID),
-			Content:    hit.Fields["Content"].(string),
+			RepoID:      int64(hit.Fields["RepoID"].(float64)),
+			StartIndex:  startIndex,
+			EndIndex:    endIndex,
+			Filename:    filenameOfIndexerID(hit.ID),
+			Content:     hit.Fields["Content"].(string),
+			CommitID:    hit.Fields["CommitID"].(string),
+			UpdatedUnix: updatedUnix,
+			Language:    language,
+			Color:       enry.GetColor(language),
+		}
+	}
+
+	languagesFacet := result.Facets["languages"]
+	searchResultLanguages := make([]*SearchResultLanguages, 0, languagesFacet.Terms.Len())
+	for _, term := range languagesFacet.Terms {
+		if len(term.Term) == 0 {
+			continue
 		}
+		searchResultLanguages = append(searchResultLanguages, &SearchResultLanguages{
```

Comment:
Does this have a weight order, like number of occurrences?

Reply:
Yes, it gives top ten languages by occurrences in desc order (first being most occurrences)